### PR TITLE
feat: move e2e tests into separate files and add new scenarios

### PR DESCRIPTION
**Pull request by cargopete**
### Description

Reached feature parity with old tests. Tests can be started using `cargo run -p test-runner`, they take about `230`s on average to complete.

### Issue link (if applicable)

[Link to issue](https://github.com/graphops/poi-radio/issues/175)

### CODE REVIEW

File: poi-radio/src/config.rs [hunk 1]
```diff
@@ -271,6 +271,12 @@ pub struct Config {
     pub radio_name: String,
     #[clap(long, value_name = "FILTER_PROTOCOL", env = "ENABLE_FILTER_PROTOCOL")]
     pub filter_protocol: Option<bool>,
+    #[clap(
+        long,
+        value_name = "TOPIC_UPDATE_INTERVAL",
+        env = "TOPIC_UPDATE_INTERVAL"
+    )]
+    pub topic_update_interval: Option<u64>,
```
Comment (hopeyen) [inline poi-radio/src/config.rs]: maybe use a default value (600) instead of unwrap later?

**Review by hopeyen — APPROVED** [review]
really like how the tests are written now, the flow is very clear!:)

File: poi-radio/src/config.rs (hunk 1 above)
Comment (cargopete) [reply poi-radio/src/config.rs]: agreed, updated

**Review by hopeyen — APPROVED** [review]
rebase and then can run tests🏃🏃🏃

**Review by hopeyen — APPROVED** [review]
lgtm! later on the mock server could take more specific server_state (such as subgraph list be different between graph node and network subgraph)

**Pull request merged**